profiling/variorum-connector/variorum-connector.cpp

```diff
@@ -14,425 +14,220 @@
 //
 //@HEADER
 
-// Modified by Zach Frye at LLNL
-// Contact: frye7@llnl.gov
-// Organization: CASC at LLNL
-
-#include <stdio.h>
-#include <inttypes.h>
-#include <cstdlib>
-#include <cstring>
-#include <vector>
-#include <unordered_set>
-#include <string>
-#include <regex>
-#include <cxxabi.h>
-#include <dlfcn.h>
-#include <ctime>
-#include <chrono>
-#include <iostream>
-#include <fstream>
+#include "kp_core.hpp"
 
-// variorum trial run
 extern "C" {
 #include <variorum.h>
 #include <jansson.h>
 }
 
-#include "kp_core.hpp"
-
-#if USE_MPI
-#include <mpi.h>
-#endif
+#include <iostream>
+#include <fstream>
 
 namespace KokkosTools {
 namespace VariorumConnector {
 
-bool filterKernels;
-uint64_t nextKernelID;
-std::vector<std::regex> kernelNames;
-std::unordered_set<uint64_t> activeKernels;
-
-typedef void (*initFunction)(const int, const uint64_t, const uint32_t, void*);
-typedef void (*finalizeFunction)();
-typedef void (*beginFunction)(const char*, const uint32_t, uint64_t*);
-typedef void (*endFunction)(uint64_t);
-
-static initFunction initProfileLibrary         = NULL;
-static finalizeFunction finalizeProfileLibrary = NULL;
-static beginFunction beginForCallee            = NULL;
-static beginFunction beginScanCallee           = NULL;
-static beginFunction beginReduceCallee         = NULL;
-static endFunction endForCallee                = NULL;
-static endFunction endScanCallee               = NULL;
-static endFunction endReduceCallee             = NULL;
-
-// the output path for ranekd output files
-std::string printpath = "./";
-
-// variables for simple timer
-time_t start_time;
+// Initial and final power values for a kernel
+double global_power[2] = {0, 0};
+// Initial and final time value for a kernel
+long long global_time[2]    = {0, 0};
+uint32_t global_device_id   = -1;
+uint32_t global_instance_id = -1;
+std::string global_filename;
+std::string global_kernel_name;
+std::string global_device_type;
```

Comment:
why do we need so many global variables?

Reply:
We need to record a bunch of information in the "begin kernel" calls that we are using in the "end kernel" calls. We can't pass that through the API interfaces directly.